### ISSUE TITLE
Some Guidance on The Why?

### ISSUE THREAD

**Issue by davidshq** [google/wwwbasic#43]
I have never lost my love for BASIC, still use QB64, and have worked with a number of other BASIC variants in the past. Of course, Google working on a BASIC variant is exciting - but it would be a bit more exciting if I (we?) understood why.

Is this _just_ a hobby language? Is it an attempt to build a new gateway into programming that can be taught in schools? Does it intend to provide compatibility with any specific variants (QB, QB64, FreeBasic, VB, etc.)?

I'd love to see the README.md or a wiki article address this topic (or a reply to this issue would be a great place to start!).

**Comment by thomasmcneill** [comment]
For me I learned to program on BASIC back when the magazines would publish
a game and you would have to type it to play.  I image something like this
can be used similarly for my son and other youth to learn to program.

**Comment by pauwell** [comment]
Same here. I started programming using Blitzbasic and this project brings back good memories. As far as I understood it is planned to provide compatability to each  variant.. or at least the important ones.

**Comment by flagxor** [comment]
Hi David,

The immediate trigger for wwwBASIC's existence was that I'd encountered Ed Thelen's Nike Hercules simulator.
It had been written in BASIC some time back, then ported to some unknown version of FreeBasic.
However, while he had screenshots and source code, it also included a Windows .EXE to a download, and the statement, "guaranteed to be free of viruses". :-/
http://ed-thelen.org/NikeSimulation.html#SimBrowser

As it was meant to capture how something historical worked, it seemed unfortunate to me that something of this sort couldn't just be accessible directly on the web. Various whole system emulators that run on the web are available, but booting a whole system for a small program seemed like overkill.

Hence the first goal was to get this to run:
http://ed-thelen.org/nike-fromBradNelsonSept26.html

From there, bringing up DONKEY.BAS seemed a nice next logical milestone.
GORILLA.BAS and NIBBLES.BAS are the current goals I have.

As for it being a Google project, 20% time is alive and well at Google, and I've been able to tinker with it a bit at the office. Google in general is very friendly towards open-sourcing projects of all sorts. While I had the option of distributing though my own github account, I figured doing it with Google's would more likely attract interested folks, and make sure I got contributor licenses right. A happily few folks have indeed shown up with contributions. :-)

BASIC being my first programming language, I can't help but suspect it has virtues for learning and more (despite its bad rap going back to Dijkstra). However, where the BASICs of the 80s were a pretty thin layer of abstraction over the hardware and offered a gateway to deep understanding, on current systems a lot of emulation / layers mean almost any of the learning languages are sadly a little bit like sandboxed toys, which I fear diminished their value as the sort of teaching tool BASIC was to me.

Part of the point of open-sourcing is to let others have say in where something goes,
so ultimately I'm happy to see what excites folks.

It's a good suggestion to capture some of this in the README, which I've now done.

Thanks

-BradN
